### PR TITLE
[keyboard] Add x16

### PR DESCRIPTION
**Pull request by yinxianwei**
## Description

- Added folder for my other keyboards

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/x16/x16.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	// runs for every action, just before processing by the firmware
+
+	return process_record_user(keycode, record);
+}
+
+void led_set_kb(uint8_t usb_led) {
+	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+	led_set_user(usb_led);
+}
```
Comment (zvecr) [inline keyboards/x16/x16.c:43]:
```suggestion
```

File: keyboards/x16/readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# x16
+
+A 16 key macropad, with USB C
+
+Keyboard Maintainer: QMK Community  
+Hardware Supported: The PCBs, controllers supported  
+Hardware Availability: The MK60 was discontinued by [x16](https://yinxianwei.github.io/x16/)
```
Comment (zvecr) [inline keyboards/x16/readme.md:7]: MK60?

File: keyboards/x16/rules.mk [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+MOUSEKEY_ENABLE = yes		# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes		# Audio control and System control(+450)
+CONSOLE_ENABLE = no	  # Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes			# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+KEY_LOCK_ENABLE = no
```
Comment (zvecr) [inline keyboards/x16/rules.mk:46]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/x16/config.h [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
```
Comment (fauxpark) [inline keyboards/x16/config.h:243]:
```suggestion
```

File: keyboards/x16/readme.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# x16
+
+A 16 key macropad, with USB C
+
+Keyboard Maintainer: QMK Community  
+Hardware Supported: The PCBs, controllers supported  
+Hardware Availability: The x16 was discontinued by [x16](https://yinxianwei.github.io/x16/)
```
Comment (fauxpark) [inline keyboards/x16/readme.md:7]:
```suggestion
* Keyboard Maintainer: QMK Community
* Hardware Supported: The PCBs, controllers supported
* Hardware Availability: The x16 was discontinued by [x16](https://yinxianwei.github.io/x16/)
```

File: keyboards/x16/rules.mk [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/x16/rules.mk:3]:
```suggestion

# Bootloader selection
```

File: keyboards/x16/rules.mk [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+# Build Options
```
Comment (fauxpark) [inline keyboards/x16/rules.mk:12]:
```suggestion

# Build Options
```

File: keyboards/x16/rules.mk [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/x16/rules.mk:27]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**